core/templates/dev/head/domain/collection/CollectionRightsObjectFactory.ts

```diff
@@ -24,78 +24,78 @@ import { Injectable } from '@angular/core';
 
 export interface ICollectionRights {
   getCollectionId: () => number;
-  canEdit: () => boolean;
-  isPrivate: () => boolean;
-  canUnpublish: () => boolean;
+  getCanEdit: () => boolean;
+  getIsPrivate: () => boolean;
+  getCanUnpublish: () => boolean;
   getOwnerNames: () => string[];
 }
 
 export class CollectionRights {
-  _collectionId: number;
-  _canEdit: boolean;
-  _canUnpublish: boolean;
-  _isPrivate: boolean;
-  _ownerNames: string[];
+  collectionId: number;
+  canEdit: boolean;
+  canUnpublish: boolean;
+  isPrivate: boolean;
+  ownerNames: string[];
 
   // TODO(#7176): Replace 'any' with the exact type. This has been kept as
   // 'any' because 'collectionRightsObject' is a dict with
   // underscore_cased keys which give tslint errors against underscore_casing
   // in favor of camelCasing.
   constructor(collectionRightsObject: any) {
-    this._collectionId = collectionRightsObject.collection_id;
-    this._canEdit = collectionRightsObject.can_edit;
-    this._canUnpublish = collectionRightsObject.can_unpublish;
-    this._isPrivate = collectionRightsObject.is_private;
-    this._ownerNames = collectionRightsObject.owner_names;
+    this.collectionId = collectionRightsObject.collection_id;
```

Comment:
Hm, looks like snake_case is still being used here? Can you please look at examples provided in the original issue or previously merged PRs that solve part of this issue so you get a clearer picture of what needs to be done? Thanks!